### PR TITLE
Bluetooth: Mesh: Add cfg cli list getters

### LINKED ISSUE #24505: Bluetooth: Mesh: Configuration Client: Add support for Model Subscription Get
**Problem**
Nodes implemented with the Configuration Client model needs to be able to send a "Config SIG Model Subscription Get" message or "Config Vendor Model Subscription Get". On the target node, the configuration server model shall reply with a "Config SIG Model Subscription List" or "Config Vendor Model Subscription List" containing all addresses from the Subscription List of the model. The Bluetooth mesh APIs does not have a function for this. 

**Possible solution**
Bluetooth mesh configuration client model:
-Add function that gets the Subscription List from target node. 
-Add model opcode handler for ("Config SIG Model Subscription List", 0x80 0x2A)
-Add model opcode handler for ("Config Vendor Model Subscription List", 0x80 0x2C)

Bluetooth mesh configuration server model:
-Add model opcode handler for("Config SIG Model Subscription Get, 0x80 0x29)
-Add model opcode handler for("Config Vendor Model Subscription Get, 0x80 0x2B)

### PR DESCRIPTION
**Pull request by trond-snekvik**
Adds config client getters for all list commands:
- mod_app_get
- mod_sub_get
- net_key_get
- app_key_get

Closes #24505.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

File: include/bluetooth/mesh/cfg_cli.h [hunk 1]
```diff
@@ -211,6 +211,21 @@ int bt_mesh_cfg_relay_set(u16_t net_idx, u16_t addr, u8_t new_relay,
...
 			    const u8_t net_key[16], u8_t *status);
 
+/** @brief Get a list of the target node's network key indexes.
+ *
+ *  @param net_idx Network index to encrypt with.
+ *  @param addr    Target node address.
+ *  @param keys    Net key index list response parameter. Will be filled with
+ *                 all the returned network key indexes it can fill.
+ *  @param key_cnt Net key index list length. Should be set to the
+ *                 capacity of the @c keys list when calling. Will return the
```
Comment (LingaoMeng) [inline include/bluetooth/mesh/cfg_cli.h:221]: @c what means?

Comment (trond-snekvik) [reply include/bluetooth/mesh/cfg_cli.h:221]: It'll make the next word render as code in the documentation

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
@robinh17 this PR should cover what you were requesting for in #24505

**Comment by robinh17** [comment]
> @robinh17 this PR should cover what you were requesting for in #24505

Thanks!

**Pull request merged**